pkg/actions/prototype_describe.go

```diff
@@ -57,6 +57,10 @@ func NewPrototypeDescribe(m map[string]interface{}) (*PrototypeDescribe, error)
 		appPrototypesFn: pkg.LoadPrototypes,
 	}
 
+	if ol.err != nil {
```

Comment:
Should we also check `ol.err` before calling any methods on the `optionLoader` above? What's the contract for `optionLoader.err`?

Reply:
No, `ol.err` would be nil if no functions were called on `ol`. `optionLoader.err` is either nil or an error. What type of contract are you looking for? Since `optionLoader` is not exported, I didn't wrap it with a function.

Reply:
A contract regarding the validity of an `optionLoader` in relation to its `err` field.

* Is it valid to call a method on an optionLoader with a non-nil `err`?
* Would all methods of `optionLoader` check `err` before their logic to protect themselves from undefined state?

I see that `newOptionLoader` can never return with a non-nil `err` field, but this is not clear to the caller except by convention.